deeplabcut/pose_estimation_3d/triangulation.py

```diff
@@ -317,67 +344,95 @@ def triangulate(
-                dataFrame_camera1_undistort, scorer_3d, flag="3d"
-            )
+
+            scorer_3d = scorer_cam1
+            bodyparts = dataFrame_camera1_undistort.columns.get_level_values(
+                "bodyparts"
+            ).unique()
+
             P1 = stereomatrix["P1"]
             P2 = stereomatrix["P2"]
+            F = stereomatrix["F"]
 
             print("Computing the triangulation...")
-            for bpindex, bp in enumerate(bodyparts):
-                # Extract the indices of frames where the likelihood of a bodypart for both cameras are less than pvalue
-                likelihoods = np.array(
-                    [
-                        dataFrame_camera1_undistort[scorer_cam1][bp][
-                            "likelihood"
-                        ].values[:],
-                        dataFrame_camera2_undistort[scorer_cam2][bp][
-                            "likelihood"
-                        ].values[:],
-                    ]
-                )
-                likelihoods = likelihoods.T
 
-                # Extract frames where likelihood for both the views is less than the pcutoff
-                low_likelihood_frames = np.any(likelihoods < pcutoff, axis=1)
-                # low_likelihood_frames = np.all(likelihoods < pcutoff, axis=1)
+            num_frames = dataFrame_camera1_undistort.shape[0]
+            ### Assign nan to [X,Y] of low likelihood predictions ###
+            # Convert the data to a np array to easily mask out the low likelyhood predictions
+            data_cam1_tmp = dataFrame_camera1_undistort.to_numpy().reshape(
+                (num_frames, -1, 3)
+            )
+            data_cam2_tmp = dataFrame_camera2_undistort.to_numpy().reshape(
+                (num_frames, -1, 3)
+            )
+            # Assign [X,Y] = nan to low likelihood predictions
+            data_cam1_tmp[data_cam1_tmp[..., 2] < pcutoff, :2] = np.nan
+            data_cam2_tmp[data_cam2_tmp[..., 2] < pcutoff, :2] = np.nan
+
+            # Reshape data back to original shape
+            data_cam1_tmp = data_cam1_tmp.reshape(num_frames, -1)
+            data_cam2_tmp = data_cam2_tmp.reshape(num_frames, -1)
+
+            # put data back to the dataframes
+            dataFrame_camera1_undistort[:] = data_cam1_tmp
+            dataFrame_camera2_undistort[:] = data_cam2_tmp
+
+            if cfg["multianimalproject"]:
+                # Check individuals are the same in both views
+                individuals_view1 = (
+                    dataFrame_camera1_undistort.columns.get_level_values("individuals")
+                    .unique()
+                    .to_list()
+                )
+                individuals_view2 = (
+                    dataFrame_camera2_undistort.columns.get_level_values("individuals")
+                    .unique()
+                    .to_list()
+                )
+                if individuals_view1 != individuals_view2:
```

Comment:
set(individuals_view1) != set(individuals_view2) would be more robust to unordered, yet matching, animal names